src/adaptors/hyperdrive/index.js

```diff
@@ -0,0 +1,324 @@
+
+          // in Hyperdrive, totalSupply and tvlUsd are the same because there is no borrowing
+          let totalSupplyUsd = ((Number(vaultSharesBalance) || 0) / 10 ** pool.config.token.decimals) * pool.config.token.price;
+          // apply vaultSharePrice from config if priceWithBase is true
+          if (pool.config.token.priceWithBase) {
+            totalSupplyUsd = totalSupplyUsd * pool.info.vaultSharePrice / 1e18;
+          }
+          let tvlUsd = Number(totalSupplyUsd) || 0;
+          let totalBorrowUsd = 0;
+
+          const result = {
+            pool: pool.name,
+            chain,
+            project: 'hyperdrive',
+            symbol: pool.config.token.symbol,
+            tvlUsd,
+            apy: apy * 100,
+            apyBase: apy * 100,
+            underlyingTokens: [pool.config.token_contract_address],
+            totalSupplyUsd,
+            totalBorrowUsd,
+            url: `app.hyperdrive.box/market/${providers[chain].chainId}/${pool.address}`
+          };
+          console.log(
+            `${chain.padEnd(10)} ${pool.name.padEnd(55)} (${pool.address.substring(0, 7)}) ` +
+            `${Math.round(tvlUsd).toLocaleString().padStart(12)} ${pool.config.token.symbol.padEnd(14)} ` +
+            `${(apy * 100).toFixed(2).padStart(5)}%`
+          );
+          return result;
+        } catch (error) {
+          console.error('Error processing pool:', pool.name, error);
+          return null;
+        }
+      })
+    );
+
+    return poolsData
+      .filter((i) => i.status === 'fulfilled')
+      .map((i) => i.value);
+  } catch (error) {
+    console.error('Error getting APY for chain:', chain, error);
+    return [];
+  }
+}
+
+async function apy() {
+  console.log('Chains to process:', Object.keys(config));
+  // Print header only once
+  console.log('network'.padEnd(10), 'pool'.padEnd(55), '(address)'.padEnd(11), 'tvl'.padStart(10), 'token'.padEnd(14), 'APR'.padStart(6));
+  const pools = await Promise.allSettled(
+    Object.keys(config).map(async (chain) => getApy(chain))
+  );
+
+  // Check for unique pool names
+  // If a duplicate is found, append [network] to its name
+  const replaceNames = {
+    xdai: "gnosis"
+  }
+  const uniquePoolNames = new Set();
+  pools
+    .filter(p => p.status === 'fulfilled')
+    .forEach(promiseResult => {
+      promiseResult.value.forEach(pool => {
+        if (pool && uniquePoolNames.has(pool.pool)) {
+          // Identify the non-ethereum pool of the pair
+          const nonEthPool = pool.chain.includes('ethereum') ? pool : pool;
+          // Add [network] to the non-ethereum pool name
+          nonEthPool.pool = `${nonEthPool.pool} [${replaceNames[nonEthPool.chain] || nonEthPool.chain}]`;
+        }
+        if (pool) {
+          uniquePoolNames.add(pool.pool);
+        }
+      });
+    });
+
+  return pools
+    .filter((i) => i.status === 'fulfilled')
+    .map((i) => i.value)
+    .flat()
+    .filter((p) => Boolean(p));
```

Comment:
shortcut you can just do p=>Boolean

Reply:
done here 74ebcc5d135198e9c915bb2d683dea0aa6a068fa